### ISSUE TITLE
SimpleFileSystem rotating_buf default doesn't work

### ISSUE THREAD

**Issue by nigelmegitt** [ebu/ebu-tt-live-toolkit#456]
Create a `filesystem-simple` output carriage with no parameters specified, and get a KeyError due to no `rotating_buf` even though there should be a default of 0.

Config file:

```json
{
  "nodes": {
    "node1": {
      "id": "producer1",
      "type": "simple-producer",
      "show_time": true,
      "sequence_identifier": "TestSequence1",
      "output": {
        "carriage": {
          "type": "direct",
          "id": "pipe1"
        }
      }
    },
    "node2": {
      "id": "resequencer1",
      "type": "resequencer",
      "sequence_identifier": "ReSequenced1",
      "segment_length": "5.0",
      "input": {
        "carriage": {
          "type": "direct",
          "id": "pipe1"
        }
      },
      "output": {
        "carriage": {
          "type": "direct",
          "id": "pipe2"
        }
      }
    },
    "node3": {
      "id": "encoder1",
      "type": "ebuttd-encoder",
      "input": {
        "carriage": {
          "type": "direct",
          "id": "pipe2"
        }
      },
      "output": {
        "carriage": {
          "type": "filesystem-simple"
        }
      }
    }
  }
}
```

Output:

```
Traceback (most recent call last):
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/venv/bin/ebu-run", line 11, in <module>
    load_entry_point('ebu-tt-live', 'console_scripts', 'ebu-run')()
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/scripts/ebu_run.py", line 42, in main
    app = create_app()
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/__init__.py", line 33, in create_app
    new_app = AppConfig(**kwargs)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/__init__.py", line 62, in __init__
    self._nodes = config.nodes.type.configure_component(config, config.nodes)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/node.py", line 370, in configure_component
    instance._nodes.append(item.type.configure_component(config, item))
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/common.py", line 53, in configure_component
    return cls(config=config, local_config=local_config)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/node.py", line 276, in __init__
    self._create_output(config)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/node.py", line 35, in _create_output
    config, self.config.output.carriage)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/common.py", line 53, in configure_component
    return cls(config=config, local_config=local_config)
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/ebu_tt_live/config/carriage.py", line 84, in __init__
    if config.rotating_buf:
  File "/Users/nigel/Code/ebu/ebu-tt-live-toolkit/venv/lib/python2.7/site-packages/configman/dotdict.py", line 363, in __getattr__
    raise KeyError(key)
KeyError: 'rotating_buf'
```

**Comment by nigelmegitt** [comment]
This looks like the same cause as #409 with the same kind of fix as in #410.

**Comment by nigelmegitt** [comment]
Plan as briefly discussed with @kozmaz87 on 2017-07-25 is to remove filesystem-simple altogether and just push any needed functionality into filesystem for output carriage stuff.
